<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1273</width>
    <height>858</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <property name="spacing">
     <number>0</number>
    </property>
    <item row="0" column="0">
     <widget class="QStackedWidget" name="stackedWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="streams">
       <layout class="QGridLayout" name="gridLayout_3">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <property name="spacing">
         <number>0</number>
        </property>
        <item row="0" column="0">
         <widget class="QWidget" name="controls" native="true">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>30</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>30</height>
           </size>
          </property>
          <widget class="QLineEdit" name="usernameEdit">
           <property name="geometry">
            <rect>
             <x>2</x>
             <y>2</y>
             <width>151</width>
             <height>26</height>
            </rect>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="placeholderText">
            <string>Username</string>
           </property>
          </widget>
          <widget class="QPushButton" name="refreshButton">
           <property name="geometry">
            <rect>
             <x>155</x>
             <y>3</y>
             <width>56</width>
             <height>24</height>
            </rect>
           </property>
           <property name="text">
            <string>Refresh</string>
           </property>
           <property name="autoDefault">
            <bool>false</bool>
           </property>
          </widget>
          <widget class="QQuickWidget" name="busyIndicator">
           <property name="geometry">
            <rect>
             <x>215</x>
             <y>5</y>
             <width>20</width>
             <height>20</height>
            </rect>
           </property>
           <property name="focusPolicy">
            <enum>Qt::NoFocus</enum>
           </property>
           <property name="resizeMode">
            <enum>QQuickWidget::SizeRootObjectToView</enum>
           </property>
           <property name="source">
            <url>
             <string>qrc:/qml/BusyIndicator.qml</string>
            </url>
           </property>
          </widget>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="StreamsScrollArea" name="scrollArea">
          <property name="frameShape">
           <enum>QFrame::NoFrame</enum>
          </property>
          <property name="verticalScrollBarPolicy">
           <enum>Qt::ScrollBarAsNeeded</enum>
          </property>
          <property name="horizontalScrollBarPolicy">
           <enum>Qt::ScrollBarAlwaysOff</enum>
          </property>
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <property name="alignment">
           <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
          </property>
          <widget class="QWidget" name="scrollAreaWidgetContents">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>1273</width>
             <height>828</height>
            </rect>
           </property>
           <layout class="QGridLayout" name="gridLayout_2">
            <property name="sizeConstraint">
             <enum>QLayout::SetNoConstraint</enum>
            </property>
            <property name="leftMargin">
             <number>0</number>
            </property>
            <property name="topMargin">
             <number>0</number>
            </property>
            <property name="rightMargin">
             <number>0</number>
            </property>
            <property name="bottomMargin">
             <number>0</number>
            </property>
            <item row="0" column="0">
             <layout class="QGridLayout" name="streamsGrid">
              <property name="sizeConstraint">
               <enum>QLayout::SetFixedSize</enum>
              </property>
              <property name="leftMargin">
               <number>3</number>
              </property>
              <property name="topMargin">
               <number>3</number>
              </property>
              <property name="rightMargin">
               <number>3</number>
              </property>
              <property name="bottomMargin">
               <number>3</number>
              </property>
              <property name="spacing">
               <number>6</number>
              </property>
             </layout>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="player">
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QQuickWidget" name="video">
          <property name="minimumSize">
           <size>
            <width>100</width>
            <height>100</height>
           </size>
          </property>
          <property name="resizeMode">
           <enum>QQuickWidget::SizeRootObjectToView</enum>
          </property>
          <property name="source">
           <url>
            <string>qrc:/qml/VideoPlayer.qml</string>
           </url>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QWebView" name="chat">
          <property name="minimumSize">
           <size>
            <width>100</width>
            <height>100</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>340</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="url">
           <url>
            <string>about:blank</string>
           </url>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QQuickWidget</class>
   <extends>QWidget</extends>
   <header>QtQuickWidgets/QQuickWidget</header>
  </customwidget>
  <customwidget>
   <class>QWebView</class>
   <extends>QWidget</extends>
   <header>QtWebKitWidgets/QWebView</header>
  </customwidget>
  <customwidget>
   <class>StreamsScrollArea</class>
   <extends>QScrollArea</extends>
   <header>StreamsScrollArea.hpp</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>usernameEdit</sender>
   <signal>returnPressed()</signal>
   <receiver>refreshButton</receiver>
   <slot>animateClick()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>77</x>
     <y>14</y>
    </hint>
    <hint type="destinationlabel">
     <x>182</x>
     <y>14</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>refreshButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>UpdateStreams()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>165</x>
     <y>15</y>
    </hint>
    <hint type="destinationlabel">
     <x>2037</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>UpdateStreams()</slot>
 </slots>
</ui>
